pgjdbc/src/test/java/org/postgresql/test/jdbc2/BatchExecuteTest.java

```diff
@@ -1405,30 +1405,23 @@ public void testMerge() throws Exception {
       stmt.executeUpdate(TestUtil.insertSQL("batchingMerge", "101, 'hello 102'"));
     }
 
+    int cnt = 16;
     try (PreparedStatement stmt = con.prepareStatement(sql)) {
-      stmt.setInt(1, 100);
-      stmt.setString(2, "");
-      stmt.addBatch();
-
-      stmt.setInt(1, 101);
-      stmt.setString(2, "");
-      stmt.addBatch();
-
-      stmt.setInt(1, 102);
-      stmt.setString(2, "hello 102");
-      stmt.addBatch();
-
-      stmt.setInt(1, 103);
-      stmt.setString(2, "hello 103");
-      stmt.addBatch();
+      int[] expected = new int[cnt];
+      for (int i = 0; i < cnt; i++) {
+        stmt.setInt(1, 100 + i);
+        stmt.setString(2, "hello " + (100 + i));
+        stmt.addBatch();
+        expected[i] = insertRewrite ? Statement.SUCCESS_NO_INFO : 1;
+      }
 
       int[] results = stmt.executeBatch();
-      int rowsInserted = insertRewrite ? Statement.SUCCESS_NO_INFO : 1;
-      Assert.assertArrayEquals(new int[]{rowsInserted, rowsInserted, rowsInserted, rowsInserted}, results);
+      Assert.assertArrayEquals(expected, results);
     }
 
-    TestUtil.assertNumberOfRows(con, "batchingMerge", 2, "2 rows expected");
+    TestUtil.assertNumberOfRows(con, "batchingMerge", cnt - 2, "14 rows expected");
```

Comment:
`14` will be a part of "expected: ... got:..." message anyway.
Please clarify in the message parameter the reasons why you expect certain things to happen.